### PR TITLE
build: add circleci config

### PR DESCRIPTION
**Pull request by mildwonkey**
We may or may not have another 0.12 release, so let's get this in place just in case.

### CODE REVIEW

**Comment by pselle** [comment]
@mildwonkey Smart. Do you think the tests might be failing because of Go versions?

**Comment by mildwonkey** [comment]
@pselle yeah, and somewhat unfortunately configuring the correct version lead to more failures 😂 I'll figure this out then re-request review, sorry for the noise!

**Comment by mildwonkey** [comment]
There we go, take two 😀

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            - cov_*.part
+
+      - store_test_results:
+          path: *TEST_RESULTS_DIR
+      - store_artifacts:
+          path: *TEST_RESULTS_DIR
+
+      - slack/status:
+          fail_only: true
+          only_for_branches: master
```
Comment (pselle) [inline .circleci/config.yml:76]: Should we add v0.12 here too?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (pselle) [inline .circleci/config.yml:106]: Same Q

File: .circleci/config.yml (hunk 1 above)
Comment (mildwonkey) [reply .circleci/config.yml:76]: I didn't see any real value in getting a notification for a branch we're not actively developing, but I have no objections to adding it.

**Pull request merged**